apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -0,0 +1,338 @@
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Info));
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+
+        [Fact]
+        public async Task Logging_TimedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<TimedBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+
+                await client.GetAsync($"/Main/Warning/{testId}");
+                await client.GetAsync($"/Main/Error/{testId}");
+
+                var noResults = GetEntries(startTime, testId, 0);
+                Assert.Null(noResults);
+                Thread.Sleep(TimeSpan.FromSeconds(25));
+
+                await client.GetAsync($"/Main/Error/{testId}");
+                await client.GetAsync($"/Main/Critical/{testId}");
+            }
+
+            // The fourth entry is not pushed as it is buffered after the last push.
+            var results = GetEntries(startTime, testId, 3);
+            Assert.Equal(3, results.Count());
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.Equal(2, results.Where(l => l.Severity == LogSeverity.Error).Count());
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="GoogleLogger"/> and associated classes.
+    /// </summary>
+    public class LoggerTestApplication
+    {
+        protected string ProjectId;
+
+        public LoggerTestApplication()
+        {
+            ProjectId = Utils.GetProjectIdFromEnvironment();
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+        }
+
+        public void SetupRoutes(IApplicationBuilder app)
+        {
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Main}/{action=Index}/{id}");
+            });
+        }
+    }
+
+    /// <summary>
+    /// An application that has a <see cref="GoogleLogger"/> with no buffer that will accept all logs
+    /// of level warning or above.
+    /// </summary>
+    public class NoBufferLoggerTestApplication : LoggerTestApplication
+    {
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+        {
+            SetupRoutes(app);
+            LoggerOptions loggerOptions = LoggerOptions.Create(LogLevel.Warning, BufferOptions.NoBuffer());
```

Comment:
It may make sense to include the log level in the application names, such as `NoBufferWarningLoggerTestApplication `, so you can have other applications names in the future with different log levels